magento/pwa-studio
# Bugfix/heuristic fragment matcher

## Description

This PR adds the `IntrospectionFragmentMatcher` to resolve the heuristic fragment matcher warning.

It:
  * Issues an introspection query to the GraphQL schema at build time to get the union and interface types
  * It passes those to the bundle via webpack's `DefinePlugin`
  * The app uses them to initialize the `IntrospectionFragmentMatcher` at run time

As such, there is also some refactoring of buildpack to make issuing GraphQL requests easier 🎉 .

## Related Issue
Related issues #1474 & #1662 


Closes #1474

## Verification Steps
1. Go to /augusta-earrings.html
2. Open chrome console
3. See console warning
4. update you code base with this PR and run  `yarn run build` in project root
5. Visit /augusta-earrings.html again and, the warning should be gone.
